wavefront_sdk/common/heartbeater_service.py

```diff
@@ -0,0 +1,63 @@
+"""
+Service that periodically reports component heartbeats to Wavefront.
+
+@author Hao Song (songhao@vmware.com).
+"""
+
+import time
+import logging
+from threading import Timer
+from wavefront_sdk.common.constants import HEART_BEAT_METRIC, \
+    APPLICATION_TAG_KEY, CLUSTER_TAG_KEY, SERVICE_TAG_KEY, SHARD_TAG_KEY, \
+    COMPONENT_TAG_KEY, NULL_TAG_VAL
+
+logging.basicConfig(level=logging.INFO)
+LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=too-few-public-methods
+class HeartbeaterService(object):
+    """Service that periodically reports component heartbeats to Wavefront."""
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, wavefront_client, application_tags, component, source,
+                 reporting_interval_seconds):
+        """
+        Construct HeartbeaterService.
+        @param wavefront_client: Wavefront Proxy or Direct Ingestion client.
+        @param application_tags: ApplicationTags.
+        @param component: Component.
+        @param source: Source.
+        @param reporting_interval_seconds: Interval of reporting heart beat.
+        """
+        self.wavefront_client = wavefront_client
```

Comment:
This one is supposed to be either WavefrontDirectClient or WavefrontProxyClient, right?

Reply:
yep